rslib/src/import_export/package/apkg/import/decks.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::{collections::HashMap, mem};
+
+use super::Context;
+use crate::{decks::NormalDeck, prelude::*};
+
+struct DeckContext<'d> {
+    target_col: &'d mut Collection,
+    usn: Usn,
+    renamed_parents: Vec<(String, String)>,
+    imported_decks: HashMap<DeckId, DeckId>,
+    unique_suffix: String,
+}
+
+impl<'d> DeckContext<'d> {
+    fn new<'a: 'd>(target_col: &'a mut Collection, usn: Usn) -> Self {
+        Self {
+            target_col,
+            usn,
+            renamed_parents: Vec::new(),
+            imported_decks: HashMap::new(),
+            unique_suffix: TimestampSecs::now().to_string(),
+        }
+    }
+}
+
+impl Context<'_> {
+    pub(super) fn import_decks_and_configs(&mut self) -> Result<HashMap<DeckId, DeckId>> {
+        let mut ctx = DeckContext::new(self.target_col, self.usn);
+        ctx.import_deck_configs(mem::take(&mut self.data.deck_configs))?;
+        ctx.import_decks(mem::take(&mut self.data.decks))?;
+        Ok(ctx.imported_decks)
+    }
+}
+
+impl DeckContext<'_> {
+    fn import_deck_configs(&mut self, mut configs: Vec<DeckConfig>) -> Result<()> {
+        for config in &mut configs {
+            config.usn = self.usn;
+            self.target_col.add_deck_config_if_unique_undoable(config)?;
+        }
+        Ok(())
+    }
+
+    fn import_decks(&mut self, mut decks: Vec<Deck>) -> Result<()> {
+        // ensure parents are seen before children
+        decks.sort_unstable_by(|d1, d2| d1.name.as_native_str().cmp(d2.name.as_native_str()));
```

Comment:
What happens if the order is wrong? A child deck is not guaranteed to match the case of its parent, and we can't use simple ASCII case folding here.

Reply:
Not so sure myself. I think a default deck would be added, and then updated when it's finally the parent's turn to be imported. Maybe we can sort by level instead.